lib/spack/spack/binary_distribution.py

```diff
@@ -45,28 +46,61 @@
 import spack.relocate as relocate
 
 
-class NoOverwriteException(Exception):
-    pass
+class NoOverwriteException(spack.error.SpackError):
 
+    def __init__(self, filepath):
+        super(NoOverwriteException, self).__init__(
+            "%s exists" % filepath)
 
-class NoGpgException(Exception):
-    pass
 
+class NoGpgException(spack.error.SpackError):
 
-class PickKeyException(Exception):
-    pass
+    def __init__(self):
+        super(NoGpgException, self).__init__(
```

Comment:
I think @tgamblin prefers using docstrings to having arg-less `__init__` methods.

Reply:
Can you point me to an example.

Reply:
`CorruptDatabaseError` in `lib/spack/spack/database.py`

Reply:
This is used in multiple places in the code. I just end up copying and pasting the error message each time I use it.

Reply:
Hmm? It should just be:

```python
class NoGpgException(spack.error.SpackError):
    """Description goes here."""
    pass
```